sources/packages/web/src/components/common/students/StudentFileUploads.vue

```diff
@@ -0,0 +1,206 @@
+<template>
+  <tab-container>
+    <body-header-container>
+      <template #header>
+        <body-header
+          title="File Uploads"
+          :recordsCount="studentFileUploads?.length"
+        >
+          <template #actions v-if="canUploadFiles">
+            <check-permission-role :role="Role.StudentUploadFile">
+              <template #="{ notAllowed }">
+                <v-btn
+                  color="primary"
+                  data-cy="uploadFileButton"
+                  @click="uploadFile"
+                  prepend-icon="fa:fa fa-plus-circle"
+                  class="float-right"
+                  :disabled="notAllowed"
+                  >Upload file</v-btn
+                >
+              </template>
+            </check-permission-role>
+          </template>
+        </body-header>
+      </template>
+      <content-group>
+        <toggle-content :toggled="!studentFileUploads.length">
+          <DataTable
+            :value="studentFileUploads"
+            :paginator="true"
+            :rows="DEFAULT_PAGE_LIMIT"
+            :rowsPerPageOptions="PAGINATION_LIST"
```

Comment:
`totalRecords`  param is also required, right?

Reply:
We are using `studentFileUploads?.length` directly.